File: flutter/lib/models/booking.dart
class Booking {
  final String? id;
  final String userId;
  final String hotelId;
  final String hotelName;
  final String roomId;
  final String roomType;
  final String checkInDate;
  final String checkOutDate;
  final int numberOfGuests;
  final int numberOfRooms;
  final int numberOfNights;
  final double totalPrice;
  final String status;
  final String? specialRequests;
  final String guestName;
  final String guestEmail;
  final String guestPhone;

  Booking({
    this.id,
    required this.userId,
    required this.hotelId,
    required this.hotelName,
    required this.roomId,
    required this.roomType,
    required this.checkInDate,
    required this.checkOutDate,
    required this.numberOfGuests,
    required this.numberOfRooms,
    required this.numberOfNights,
    required this.totalPrice,
    required this.status,
    this.specialRequests,
    required this.guestName,
    required this.guestEmail,
    required this.guestPhone,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['id'],
      userId: json['userId'],
      hotelId: json['hotelId'],
      hotelName: json['hotelName'],
      roomId: json['roomId'],
      roomType: json['roomType'],
      checkInDate: json['checkInDate'],
      checkOutDate: json['checkOutDate'],
      numberOfGuests: json['numberOfGuests'],
      numberOfRooms: json['numberOfRooms'],
      numberOfNights: json['numberOfNights'],
      totalPrice: json['totalPrice'].toDouble(),
      status: json['status'],
      specialRequests: json['specialRequests'],
      guestName: json['guestName'],
      guestEmail: json['guestEmail'],
      guestPhone: json['guestPhone'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'userId': userId,
      'hotelId': hotelId,
      'hotelName': hotelName,
      'roomId': roomId,
      'roomType': roomType,
      'checkInDate': checkInDate,
      'checkOutDate': checkOutDate,
      'numberOfGuests': numberOfGuests,
      'numberOfRooms': numberOfRooms,
      'numberOfNights': numberOfNights,
      'totalPrice': totalPrice,
      'status': status,
      if (specialRequests != null) 'specialRequests': specialRequests,
      'guestName': guestName,
      'guestEmail': guestEmail,
      'guestPhone': guestPhone,
    };
  }
}

File: backend/src/main/java/com/hotelbooker/booking/entity/Booking.java
package com.hotelbooker.booking.entity;

import com.hotelbooker.auth.entity.User;
import com.hotelbooker.common.entity.BaseEntity;
import com.hotelbooker.hotel.entity.Hotel;
import com.hotelbooker.hotel.entity.Room;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "bookings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Booking extends BaseEntity {
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hotel_id", nullable = false)
    private Hotel hotel;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id", nullable = false)
    private Room room;
    
    @Column(nullable = false)
    private LocalDate checkInDate;
    
    @Column(nullable = false)
    private LocalDate checkOutDate;
    
    @Column(nullable = false)
    private Integer numberOfGuests;
    
    @Column(nullable = false)
    private Integer numberOfRooms;
    
    @Column(nullable = false)
    private Integer numberOfNights;
    
    @Column(nullable = false)
    private Double totalPrice;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BookingStatus status = BookingStatus.PENDING;
    
    private String specialRequests;
    
    private String guestName;
    private String guestEmail;
    private String guestPhone;
    
    public enum BookingStatus {
        PENDING,
        CONFIRMED,
        CANCELLED,
        COMPLETED
    }
}
